ollama_obsidian_indexer/llm.py

```diff
@@ -101,8 +101,11 @@ def directory_reader_setup(is_dir, path):
 
 
 def index_dir(dir_path):
-    parser = directory_reader_setup(True, dir_path)
-    return nodes_to_vector(parser.load_data())
+    if any(fname.endswith(".md") for fname in os.listdir(dir_path)):
+        parser = directory_reader_setup(True, dir_path)
+        return nodes_to_vector(parser.load_data())
+    else:
+        return "No markdown files found in the directory"
```

Comment:
This should return a number, we can probably assign -1 return value and map it in the server.py to this message

Reply:
@brumik Sounds good, I think I looked at `nodes_to_vector`'s `"Vector Created"` return for an example of existing behavior, but a number is good. Fixed with my most recent commit to this branch. 

Maybe it would make sense to change these to return a tuple of `(int: # number indexed, str: error or explanation)`  In that case you could reserve -1 for actual errors, but keep distinct return values for non-error reasons for 0 files being indexed. 

Mostly just brainstorming. Thanks for the feedback, hope this PR helps!

Reply:
That would make sense, I know it is mostly a happy path app since the focus was on getting it out asap. We can definitely do this. Maybe we can open an issue and we can work on it in a new PR?

Reply:
Sounds good to me, I know you're trying to get indexing moved into the main plugin, I actually really like that it's separate, maybe something to consider allowing as a configurable option once you do move it over. Of course then you have to continue to maintain it so... may not be worth it

Reply:
I could not do that reliably, so I expect maintaining this backend part. I feel like it could be a stripped down version, but then we have no room for improvement and it could be really slow in general. You still need a machine with Ollama running on it. Only thing I do not know how to solve with this Python code is different users/vaults.